### PR TITLE
[AC-2915] Upgrade Stripe.net to 45.7.0

### PR DESCRIPTION
**Pull request by amorask-bitwarden**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/AC-2915

## 📔 Objective

This PR upgrades [Stripe.net](https://github.com/stripe/stripe-dotnet) in `Core` from `v43.20.0` to `v45.7.0`.

Additionally, because this major version bump includes an update to Stripe's latest [API version](https://docs.stripe.com/api/versioning) (2024-06-20), it also adds a new configuration to the `BillingSettings`: `StripeWebhookSecret20240620`. This setting will contain the webhook secret for the newly created webhooks using version 2024-06-20. It also makes sure to handle that version when picking a Stripe webhook secret to use in the webhook processor.

Finally, it updates our test JSON files in the Resources folder to use the newer API version as well.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Review by cyprain-okeke — DISMISSED** [review]

File: src/Billing/BillingSettings.cs [hunk 1]
```diff
@@ -6,6 +6,7 @@ public class BillingSettings
     public virtual string StripeWebhookKey { get; set; }
     public virtual string StripeWebhookSecret { get; set; }
```
Comment (cturnbull-bitwarden) [inline src/Billing/BillingSettings.cs:7]: This can be part of another PR, but we can go ahead with deleting this oldest secret, `StripeWebhookSecret`.

**Review by cturnbull-bitwarden — DISMISSED** [review]
Looks good!

**Review by cturnbull-bitwarden — APPROVED** [review]
